src/SFML/System/String.cpp

```diff
@@ -106,15 +106,15 @@ String::String(const std::wstring& wideString)
 
 
 ////////////////////////////////////////////////////////////
-String::String(const std::uint32_t* utf32String)
+String::String(const char32_t* utf32String)
 {
     if (utf32String)
         m_string = utf32String;
 }
 
 
 ////////////////////////////////////////////////////////////
-String::String(const std::basic_string<std::uint32_t>& utf32String) : m_string(utf32String)
+String::String(const std::u32string& utf32String) : m_string(utf32String)
 {
 }
```

Comment:
I think there's a clang-tidy check for that very thing if you're interested in adding it. I can do that in a later PR.

https://clang.llvm.org/extra/clang-tidy/checks/performance/move-constructor-init.html

Reply:
Ah, interesting. From the description:

> The check flags user-defined move constructors that have a ctor-initializer initializing a member or base class through a copy constructor instead of a move constructor.

That wouldn't fire here though, as this is not a move constructor, or would it? Might need testing...